code/__DATASTRUCTURES/stacks.dm

```diff
@@ -0,0 +1,55 @@
+
+//Both count as failures, and they don't equate to each other
+//this lets us do if(pop()) without having to specifically check for underflow
+//same for if(push()) and overflow
+#define STACK_OVERFLOW	""
+#define STACK_UNDERFLOW	null
+
+
+/datum/stack
```

Comment:
Tbh, Datastructures > Material stacks (that logically should just get called ```material_stack``` anyways :P)